src/rf/apps/workers/process.py

```diff
@@ -242,9 +250,45 @@ def validate_image(self, record):
                 return True
         return False
 
-    def thumbnail(self, record):
+    def thumbnail_image(self, record):
         """
-        Generate thumbnails for all images.
+        Generate thumbnails for an image.
+        record -- attribute data from SQS.
+        """
+        try:
+            data = record['data']
+            image_id = data['image_id']
+        except KeyError:
+            return False
+
+        try:
+            image_id = int(image_id)
+        except ValueError:
+            return False
+
+        image = LayerImage.objects.get(id=image_id)
+        layer_id = image.layer.id
+        log.info('Generating thumbnails for image %d...', image_id)
+        status_updates.mark_image_status_start(image.s3_uuid,
+                                               enums.STATUS_THUMBNAIL)
+
+        if make_thumbs_for_layer_image(image_id):
+            status_updates.mark_image_status_end(image.s3_uuid,
+                                                 enums.STATUS_THUMBNAIL)
+            log.info('Done generating thumbnails for image %d...', image_id)
+            status_updates.update_thumbnail_status(layer_id)
+            return True
+        else:
+            log.info('Failed to thumbnail image %d', image_id)
+            status_updates.mark_image_status_end(
+                image_id,
+                enums.STATUS_THUMBNAIL,
+                ERROR_MESSAGE_THUMBNAIL_FAILED)
+            return False
```

Comment:
Actually you probably want to return True here. The image failed to thumbnail but you don't want to re-queue it necessarily do you?